Documentation/proposals/202304-silences.md

```diff
@@ -0,0 +1,136 @@
+
+![CI/CD K8s Job Approach](../img/CICD-k8s-job.png "CI/CD K8s Job Approach")
+
+* There are few folks that stick in creating silences manually in Alertmanager UI, which means that developers need access to alertmanagers
+  and add silences manually, but again, there's no reconciliation loop.
+
+* Users that use a standalone operator that implements a Silence CRD are in a better situation,
+  for example, users have reported that they are using GitOps to fully manage the life cycle of Silences Custom resources,
+  which in fact brings benefits to the whole team because having a history and revisions in Git as well as constancy to manage the silences in a multi-tenant environment.
+
+Additionally, having a new component in the stack and keeping it maintained is not always ideal (said the folks at [Giant Swarm](https://giantswarm.io) the owners of [silence-operator](https://github.com/giantswarm/silence-operator)),
+because that requires managing an extra component +CRDs on top of the Prometheus operator.
+Moreover, this project doesn't have a lot of community support, and it's reached by small amount of users.
+Having the Silence CRD as part of prometheus-operator provides users with a better experience: there is no need to install another operator or build a system on top of it to provide that functionality.
+
+Adding support for `Silence` CRD in the Prometheus-operator will provide more flexibility to users in terms of choosing the tool to deploy their silences and will free users from managing a standalone component within the stack.
+
+## Pitfalls of the current solution
+
+Using Alertmanager API Directly comes with drawbacks:
+
+* Teams have to build an automation to add silences in a centralized manner
+* There is no input validation in CI, which can lead to an invalid silence configuration
+* Teams need to manage expiration on their own
+* If a silence was deleted by accident, it will be permanent.
+
+# Goals
+
+* Provide a way for users to manage Alertmanager silences with a Custom Resource Definition (CRD).
+* Enable management of Alertmanager silences via centralised repo with e.g. ArgoCD, Flux...
+* Multi tenancy: Silences CR can be using kubernetes rbac.
+* Provide better community support and reduce fragmentation as Giant Swarm's silence operator are added to prometheus-operator
+* Improve user experience when managing silences, so users can use the same tools to manage their monitoring pipeline (kyverno, cr validations).
+
+## Audience
+
+* Users who use the Prometheus operator to offer monitoring-as-a-service to developers.
+* Users who want to manage silences the same way as for services running within the Kubernetes cluster.
+* Users who want a supported GitOps way of silences outside the Kubernetes cluster
+
+# Non-Goals
+
+* This proposal do not cover automatic cleanup of expired Silence CRs
+* Refactoring of the other CRDs is not in scope for the first version
+* Adding silences to a remote Alertmanager cluster
```

Comment:
not sure to understand this one. Is it about creating silences on an Alertmanager cluster not managed by the operator?

Reply:
yes, we will not cover adding silences to a remote alertmanager, at least not in the same cluster.

Reply:
It can actually be easily achieved with optional field in the CRD for alertmanager url/auth/etc.

If it's not provided operator would take default one from config